bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,365 @@
+    future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .publish_with_options(options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()
+    })
+  }
+
+  /// Signs a {@link Credential} with the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedCredential)]
+  pub fn create_signed_credential(
+    &self,
+    fragment: String,
+    credential: &WasmCredential,
+    signature_options: WasmSignatureOptions,
+  ) -> PromiseCredential {
+    let account = self.0.clone();
+    let mut cred: Credential = credential.0.clone();
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut cred, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&cred).wasm_result()
+    });
+    promise.unchecked_into::<PromiseCredential>()
+  }
+
+  /// Signs a {@link Document} with the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedDocument)]
+  pub fn create_signed_document(
+    &self,
+    fragment: String,
+    document: &WasmDocument,
+    signature_options: WasmSignatureOptions,
+  ) -> PromiseDocument {
+    let account = self.0.clone();
+    let mut doc: IotaDocument = document.0.clone();
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut doc, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&doc).wasm_result()
+    });
+    promise.unchecked_into::<PromiseDocument>()
+  }
+
+  /// Signs a {@link Presentation} the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedPresentation)]
+  pub fn create_signed_presentation(
+    &self,
+    fragment: String,
+    presentation: &WasmPresentation,
+    signature_options: WasmSignatureOptions,
+  ) -> PromisePresentation {
+    let account = self.0.clone();
+    let mut pres: Presentation = presentation.0.clone();
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    wasm_logger::init(wasm_logger::Config::default());
+    log::info!("{:?}", options);
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut pres, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&pres).wasm_result()
+    });
+    promise.unchecked_into::<PromisePresentation>()
+  }
+
+  /// Signs arbitrary `data` with the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedData)]
+  pub fn create_signed_data(
```

Comment:
This should be called `signData` because it has the same behaviour as `WasmDocument::signData`, developers will not expect a difference.

Reply:
Point still stands I think.